src/main/java/cd/go/framework/ldap/Ldap.java

```diff
@@ -99,15 +101,23 @@ private static SearchControls getSimpleSearchControls(int maxResult) {
 
     private NamingEnumeration<SearchResult> search(String filter, Object[] filterArgs, int maxResult) throws NamingException {
         DirContext dirContext = getDirContext(ldapConfiguration, ldapConfiguration.getManagerDn(), ldapConfiguration.getPassword());
-        for (String base : ldapConfiguration.getSearchBases()) {
-            try {
+        try {
+            for (String base : ldapConfiguration.getSearchBases()) {
```

Comment:
`NamingEnumeration<SearchResult>` or `SearchResult` is not having ldap connection reference, this is fine